### DISCUSSION TITLE
「开源之史」系列之十三/四：如何限制基于网络服务的闭源？从AGPL到SSPL，开源许可正在失控？ - 「开源之道」

### DISCUSSION THREAD

**Discussion by giscus** [OCselected/ocselected.github.io#66] [Announcements]
# 「开源之史」系列之十三/四：如何限制基于网络服务的闭源？从AGPL到SSPL，开源许可正在失控？ - 「开源之道」

https://opensourceway.community/posts/history-of-open-source/02-09-cloud-computer-and-new-non-os-license/

**Comment by ZoomQuiet** [comment]
#是也乎,(￣▽￣)

这种情况其实, 就是另外一篇的高级版本:
[在开源的世界里如何应对搭便车者？ \- 「开源之道」](https://opensourceway.community/posts/the_way_of_open_source/how-to-repose-freerider-in-open-source/)

搭便车客/ freerider  其实是有门槛的,
原本 FLOSS/自由开源软件 之所以创造出专有许可证, 
就是为了令软件自由, 自己以及同好们的智力创作可以永久开放为人类进步作出贡献, 
而不是仅仅变成少数组织牟利的工具;

以往 freerider 只是少数没良心的技术人员,
认为 开源就是免费, 你们圣母, 活该免费给我们服务;

可现在 SaaS 厂商, 则更进一步, 通过巨大的资源投入,
完全消化 FLOSS 成果,
提升自己的系统品质, 节省海量成本的同时, 
并没对等反馈应该的改进...

可问题在, 厂商们这么作, 并不是良心坏了, 而是企业的必然选择,
企业这种生物只为一个目的而挣扎 -- 更多利润;
凡是减少利润的要求都将遭受最大的恶意;

FLOSS 想在云厂商黑洞前生存,
只能通过法律手段了,
这不象以往, 软件厂商总是有很多同类竞争对手, 其中总是有脑洞大开的,
通过拥抱开源而获得新生;

可云厂商不是, 云厂商是趋向垄断的, 最后可能只有一个云厂商, 无比强大的云平台,
吸收并能完全吸收一切开源成果,
并坚定的不反馈任何改进,
此时怎么办?

一般问题解决思路就三个方向:

+ 正怼: 既然争议在许可证, 那么升级, 并联合升级许可证, 要求法律来要求厂商回到开源循环中
+ 反挖: 厂商不傻, 法律也只是工具, 从技术角度如何限制云厂商不正义的使用? 如果能有技术手段识别云环境和本地运行? 是否可以追加上云锁, 来迫使厂商合理支付?
+ 别途: 云不应该是一切的终点, 如果云原生生性如此, 那么, 值得开展其它形态互联网的探索了, 如果局域网组成的应用也可以形成品质不弱于云应用的系统, 那么, 开源开始关注这类生态, 来对抗死星一般的云厂商呢?
+ ...

**Reply by tisonkun** [reply]
> 堂而皇之的违反GPL条款

违反 GPL 精神？既然是 GPL 的 Loophole，那自然没有”违反“一说。

**Reply by tisonkun** [reply]
噗，发成回复了。

**Reply by lijiangsheng1** [reply]
感谢反馈，已更改。

**Comment by tisonkun** [comment]
> 云计算巨头 AWS 则 fork 了这个项目，发布了兼容版本的 DocumentDB

这个表述不准确吧？AWS DocumentDB 确实做了 MongoDB 的兼容，但是并不是 fork。

**Reply by lijiangsheng1** [reply]
嗯，AWS DocumentDB 是产品名，不是开源的项目。 笔误。
